src/cargo/util/config/mod.rs

```diff
@@ -545,9 +561,27 @@ impl Config {
                 return true;
             }
         }
+        self.check_environment_key_case_mismatch(key);
+
         false
     }
 
+    fn check_environment_key_case_mismatch(&self, key: &ConfigKey) {
+        if cfg!(windows) {
+            return;
```

Comment:
I think it might be good to have a comment here explaining why this is ignored on windows (for those who may not be familiar that windows automatically ASCII upper-cases keys).

(Hopefully nobody uses lower-case non-ASCII characters 😄.)